MU-1614 Write an example on how to test a Flow with a scheduler (#121)
mulesoft/docs-munit: modules/ROOT/pages/test-flow-with-scheduler-cookbook.adoc

= Testing a Flow with a Scheduler Source
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Assume that you want to test the following flow triggered by an Scheduler component, and that you want to validate that your flow is being triggered correctly:

Reviewer: ```suggestion
Assume that you want to both test the following flow triggered by a Scheduler component and validate that your flow is being triggered correctly:
```

[source,xml]
----
<flow name="schedulerFlow">
    <scheduler>
        <scheduling-strategy >
            <fixed-frequency frequency="10" timeUnit="SECONDS"/>
        </scheduling-strategy>
    </scheduler>
    <set-payload value="#['test poll']"/>
</flow>
----

> . Enable Flow Sources. +

Reviewer: ```suggestion
. Enable flow sources. +
```

> By default, MUnit does not start any of the event sources (flow sources such as triggers or listeners). You must configure which flow sources must start for your tests.

Reviewer: Why introduce the synonym "event sources" here if we are otherwise referring to them as "flow sources"?
Using synonyms runs the risk of confusing audiences, who might assume that different wordings (for one thing) equates to different meanings.

Reviewer: It might be simpler to reword to:
```suggestion
Because MUnit does not start any of the flow sources (such as triggers or listeners), you must configure which flow sources start for your tests.
```

+

> In this case, you must enable the flow that contains the Scheduler:

Reviewer: ```suggestion
In this example, you enable the flow that contains the Scheduler:
```

+
[source,xml]
----
<munit:test name="test-schedulerFlowTest">
    <munit:enable-flow-sources>
        <munit:enable-flow-source value="schedulerFlow"/>
    </munit:enable-flow-sources>
</munit:test>
----

> . Verify Flow Execution is Triggered by the Scheduler. +

Reviewer: ```suggestion
. Verify that flow execution is triggered by the Scheduler. +
```

> To check that the processors inside your flow are being triggered, use the `munit-tools:verify-call` processor to assert that any of the processors was called.

Reviewer: ```suggestion
To verify that the processors inside your flow are being triggered, use the `munit-tools:verify-call` processor.
```

+

> Wrap the `munit-tools:verify-call` inside an `until-successful` router because the Scheduler may not be executed at that particular time and you need to stay checking for a shorter time.

Reviewer: ```suggestion
Wrap `munit-tools:verify-call` inside an `until-successful` to shorten processing time when the Scheduler is not operating when you want to test.
```

+

> The following example checks that the payload was executed at least one time. It validates this condition three times to ensure that the flow was executed at the time the test runs.

Reviewer: "that the flow was executed" or "that the payload was executed"?

Is it confusing that the payload was executed at least once but the validation occurred three times?

Does this mean the same thing?
```suggestion
The following example validates three times that the payload was executed at least once during the test. 
```
